libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -2284,7 +2284,7 @@ ChannelSampleView WaveTrack::GetOneSampleView(
       //    t0 - clipT0 <= 0,
       // then `length >= len`. `as_size_t` won't narrow.
       const auto len = TimeToLongSamples(t1 - clipT0).as_size_t();
-      auto newSegment = clip->GetSampleView(0u, clipS0, len);
+      auto newSegment = clip->GetSampleView(clipS0, len)[0u];
```

Comment:
This call no longer needs to satisfy the removed precondition, but must now not index the array out of bounds, and so `ClipInterface::GetSampleView` should have a `@post` or `@return` comment guaranteeing the size of the result equals the width.

And `ClipInterface::GetWidth()` should replicate the promise of `WideSampleSequence::NChannels`:

```
   //! A constant property
   /*!
    @post result: `result > 0`
    */
```